Week4/app/src/main/java/com/oliverperez/blockbusterchallenge/Models/Movie.kt

```diff
@@ -0,0 +1,46 @@
+package com.oliverperez.blockbusterchallenge.Models
+
+import android.os.Parcel
+import android.os.Parcelable
+import java.util.*
+
+/*
+* Data class to represent a Movie item abstraction
+*
+* @property id The unique ID of the movie
+* @property releaseDate The date in which the film was released
+* @property title The title of the movie
+* @property summary The summary of the movie
+* @property poster The poster image of the movie
+ */
+data class Movie(val id: Int,
```

Comment:
You can use `@Parcelize` annotation on a class to generate entire parcelable implementation.

```kotlin
@Parcelize
data class Movie
```